include/my_rdtsc.h

```diff
@@ -175,6 +179,28 @@ static inline ulonglong my_timer_cycles(void)
     __asm __volatile("mrs	%0, CNTVCT_EL0" : "=&r" (result));
     return result;
   }
+#elif defined(__riscv)
+  /* Use RDCYCLE (and RDCYCLEH on riscv32) */
+  {
+# if __riscv_xlen == 32
+    ulong result_lo, result_hi0, result_hi1;
+    /* Implemented in assembly because Clang insisted on branching. */
+    __asm __volatile__(
+        "rdcycleh %0\n"
+        "rdcycle %1\n"
+        "rdcycleh %2\n"
+        "sub %0, %0, %2\n"
+        "seqz %0, %0\n"
+        "sub %0, zero, %0\n"
+        "and %1, %1, %0\n"
+        : "=r"(result_hi0), "=r"(result_lo), "=r"(result_hi1));
+    return (static_cast<ulonglong>(result_hi1) << 32) | result_lo;
```

Comment:
I think using `static_cast` here is problematic because this header is included in `mysys/my_rdtsc.c` which is compiled with a C compiler not C++ compiler, so it ends up complaining about this

```
include/my_rdtsc.h:229:25: error: unexpected type name 'ulonglong': expected expression
  229 |     return (static_cast<ulonglong>(result_hi1) << 32) | result_lo;
include/my_rdtsc.h:229:13: error: use of undeclared identifier 'static_cast'
  229 |     return (static_cast<ulonglong>(result_hi1) << 32) | result_lo;
      |             ^
```

I I change this to 

```
return ((ulonglong)result_hi1 << 32) | result_lo;
```

it works